Framework/src/MonitorObjectCollection.cxx

```diff
@@ -46,16 +45,19 @@ void MonitorObjectCollection::merge(mergers::MergeInterface* const other)
       if (!otherMO || !targetMO) {
         throw std::runtime_error("The target object or the other object could not be casted to MonitorObject.");
       }
-      if (!reportedMismatchingRunNumbers && targetMO->getActivity().mId != otherMO->getActivity().mId) {
+
+      if (otherMO->getActivity().mId > targetMO->getActivity().mId) {
         ILOG(Error, Ops) << "The run number of the input object '" << otherMO->GetName() << "' ("
                          << otherMO->getActivity().mId << ") "
-                         << "does not match the run number of the target object '"
+                         << "is higher than the one of the target object '"
                          << targetMO->GetName() << "' (" << targetMO->getActivity().mId
-                         << "). Trying to continue, but THIS SHOULD BE IMMEDIATELY ADDRESSED IN PRODUCTION. "
-                         << "QC objects from other setups are reaching this one. Will not report more mismatches in this collection."
+                         << "). I am switching these objects, but THIS SHOULD BE IMMEDIATELY ADDRESSED IN PRODUCTION. "
```

Comment:
"I am switching these objects" is not clear to me, how about "Replacing the merged object with input"?